### ISSUE TITLE
[Bug]: npm could not resolve dependency

### ISSUE THREAD

**Issue by sergey-belous** [ChrisUser/cra-template-complete-web-app#3]
### What happened?

Can not create new project from `npx create-react-app new-test-app --template complete-web-app`
Could not resolve dependency.

### Version

v1.1.15

### Relevant log output

```shell
Creating a new React app in /home/sj/vscode/test-react-app/new-test-app.

Installing packages. This might take a couple of minutes.
Installing react, react-dom, and react-scripts with cra-template-complete-web-app...


added 1496 packages in 2m

259 packages are looking for funding
  run `npm fund` for details

Initialized a git repository.

Installing template dependencies using npm...
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: new-test-app@0.1.0
npm WARN Found: typescript@5.4.2
npm WARN node_modules/typescript
npm WARN   peer typescript@">= 2.7" from fork-ts-checker-webpack-plugin@6.5.3
npm WARN   node_modules/fork-ts-checker-webpack-plugin
npm WARN     fork-ts-checker-webpack-plugin@"^6.5.0" from react-dev-utils@12.0.1
npm WARN     node_modules/react-dev-utils
npm WARN   2 more (tsutils, the root project)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peerOptional typescript@"^3.2.1 || ^4" from react-scripts@5.0.1
npm WARN node_modules/react-scripts
npm WARN   react-scripts@"5.0.1" from the root project

added 413 packages, removed 31 packages, and changed 41 packages in 44s

271 packages are looking for funding
  run `npm fund` for details

Removing template package using npm...

npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: react-scripts@5.0.1
npm ERR! Found: typescript@5.4.2
npm ERR! node_modules/typescript
npm ERR!   peer typescript@">= 2.7" from fork-ts-checker-webpack-plugin@6.5.3
npm ERR!   node_modules/fork-ts-checker-webpack-plugin
npm ERR!     fork-ts-checker-webpack-plugin@"^6.5.0" from react-dev-utils@12.0.1
npm ERR!     node_modules/react-dev-utils
npm ERR!       react-dev-utils@"^12.0.1" from react-scripts@5.0.1
npm ERR!       node_modules/react-scripts
npm ERR!         react-scripts@"^5.0.1" from the root project
npm ERR!   peer typescript@">=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta" from tsutils@3.21.0
npm ERR!   node_modules/tsutils
npm ERR!     tsutils@"^3.21.0" from @typescript-eslint/eslint-plugin@5.62.0
npm ERR!     node_modules/@typescript-eslint/eslint-plugin
npm ERR!       @typescript-eslint/eslint-plugin@"^5.5.0" from eslint-config-react-app@7.0.1
npm ERR!       node_modules/eslint-config-react-app
npm ERR!         eslint-config-react-app@"^7.0.1" from react-scripts@5.0.1
npm ERR!         node_modules/react-scripts
npm ERR!       1 more (eslint-plugin-jest)
npm ERR!     tsutils@"^3.21.0" from @typescript-eslint/type-utils@5.62.0
npm ERR!     node_modules/@typescript-eslint/type-utils
npm ERR!       @typescript-eslint/type-utils@"5.62.0" from @typescript-eslint/eslint-plugin@5.62.0
npm ERR!       node_modules/@typescript-eslint/eslint-plugin
npm ERR!         @typescript-eslint/eslint-plugin@"^5.5.0" from eslint-config-react-app@7.0.1
npm ERR!         node_modules/eslint-config-react-app
npm ERR!         1 more (eslint-plugin-jest)
npm ERR!     1 more (@typescript-eslint/typescript-estree)
npm ERR!   1 more (the root project)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peerOptional typescript@"^3.2.1 || ^4" from react-scripts@5.0.1
npm ERR! node_modules/react-scripts
npm ERR!   react-scripts@"^5.0.1" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: typescript@4.9.5
npm ERR! node_modules/typescript
npm ERR!   peerOptional typescript@"^3.2.1 || ^4" from react-scripts@5.0.1
npm ERR!   node_modules/react-scripts
npm ERR!     react-scripts@"^5.0.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /home/sj/.npm/_logs/2024-03-17T09_21_16_712Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ChrisUser** [comment]
Hi @sergey-belous, sorry for the late reply.

I've published a new release ([v1.1.16](https://github.com/ChrisUser/cra-template-complete-web-app/commit/76a8ddfec8de83e3a2fb5553d042b628cedb3f52)) that includes a fix described in https://github.com/facebook/create-react-app/issues/13080#issuecomment-1515280885. Let me know if it solves all your issues.

I'll soon publish a new template for a complete React app configuration using Vite as suggested by the React community.
Sadly Create React App seems like a [dead project](https://github.com/facebook/create-react-app/issues/13393), but I won't stop the support for this template yet.

**Comment by ChrisUser** [comment]
Going to close this one due to inactivity. Feel free to re-open in case the problem recurs.
